mne/time_frequency/psd.py

```diff
@@ -11,17 +11,19 @@
 from .multitaper import _psd_multitaper
 
 
-def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func):
+def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func, padding):
     """Aux function."""
     return func(epoch, fs=fs, nperseg=nfft, noverlap=noverlap,
-                nfft=nfft, window='hann')[2][..., freq_mask, :]
+                nfft=padding, window='hann')[2][..., freq_mask, :]
```

Comment:
This is the scipy.spectral.spectrogram API. It differs from mne's psd_welch: `nperseg` is the length of each welch window and `nfft` is the final length of each window. So if nfft > nperseg it results in zero padding nperseg segments to nfft. psd_welch already uses nfft to mean nperseg, so I added padding (or `pad_to` or a different kwarg name if you prefer). We could also change psd_welch a little to use the same logic as scipy spectrogram (nperseg and nfft) but that would require deprecation (although the behavior could stay backward compatible - the meaning of the parameters would change).